components/core/src/clp/Grep.cpp

```diff
@@ -1063,4 +1011,437 @@ size_t Grep::search(Query const& query, size_t limit, Archive& archive, File& co
+        // one-to-one, wildcard variables that may be encoded have different logtype strings when
+        // comparing against the dictionary than they do when comparing against the segment.
+        std::string logtype_string;
+        bool has_vars = true;
+        for (uint32_t i = 0; i < query_logtype.get_logtype_size(); i++) {
+            auto const& logtype_value = query_logtype.get_logtype_value(i);
+            auto const& raw_string = query_logtype.get_query_string(i);
+            auto const& is_dict_var = query_logtype.get_is_potentially_in_dict(i);
+            auto const& var_has_wildcard = query_logtype.get_has_wildcard(i);
+            if (std::holds_alternative<char>(logtype_value)) {
+                logtype_string.push_back(std::get<char>(logtype_value));
+            } else {
+                auto& schema_type = lexer.m_id_symbol[std::get<int>(logtype_value)];
+                encoded_variable_t encoded_var;
+
+                // If this logtype contains wildcard variables that are being compared against the
+                // dictionary, create a duplicate logtype that will compare against segment as the
+                // variable may be encoded there instead.
+                if (false == is_dict_var && var_has_wildcard
+                    && (schema_type == "int" || schema_type == "float"))
+                {
+                    QueryLogtype new_query_logtype = query_logtype;
+                    new_query_logtype.set_var_is_potentially_in_dict(i, true);
+                    // TODO: sketchy, but works cause < operator inserts it after current iterator
+                    query_logtypes.insert(new_query_logtype);
+                }
+                if (is_dict_var) {
+                    if (schema_type == "int") {
+                        LogTypeDictionaryEntry::add_int_var(logtype_string);
+                    } else if (schema_type == "float") {
+                        LogTypeDictionaryEntry::add_float_var(logtype_string);
+                    }
+                } else if (schema_type == "int"
+                           && EncodedVariableInterpreter::
+                                   convert_string_to_representable_integer_var(
+                                           raw_string,
+                                           encoded_var
+                                   ))
+                {
+                    LogTypeDictionaryEntry::add_int_var(logtype_string);
+                } else if (schema_type == "float"
+                           && EncodedVariableInterpreter::convert_string_to_representable_float_var(
+                                   raw_string,
+                                   encoded_var
+                           ))
+                {
+                    LogTypeDictionaryEntry::add_float_var(logtype_string);
+                } else {
+                    LogTypeDictionaryEntry::add_dict_var(logtype_string);
+                }
```

Comment:
I think this logic needs a bit of reworking:

* If a variable matches an `int` or `float` AND has a wildcard, then trying to encode it as an `int` or `float` will fail (since those methods don't expect wildcards). However, we should still treat them as `int` or `float` when adding them to the logtype.
* Instead of tracking whether a token could be a dictionary variable, I think it's clearer to track whether it could be an encoded variable. So the idea would be:
  * if it's an encoded variable, we try to encode it (if it doesn't have a wildcard) etc.
  * else we treat it as a dictionary variable and if it can be an encoded variable, we add new query logtypes to handle those possibilities.